src/chocolatey/infrastructure.app/nuget/NugetCommon.cs

```diff
@@ -139,19 +148,50 @@ public static IPackageRepository GetRemoteRepository(ChocolateyConfiguration con
 
                 try
                 {
+                    IPackageRepository packageRepository;
                     var uri = new Uri(source);
                     if (uri.IsFile || uri.IsUnc)
                     {
-                        repositories.Add(new ChocolateyLocalPackageRepository(uri.LocalPath) { Logger = nugetLogger });
+                        packageRepository = new ChocolateyLocalPackageRepository(uri.LocalPath) { Logger = nugetLogger };
+                    }
+                    else
+                    {
+                        packageRepository = new DataServicePackageRepository(new RedirectedHttpClient(uri, bypassProxy) { UserAgent = "Chocolatey Core" }, packageDownloader) { Logger = nugetLogger };
+                    }
+
+                    // The priority should never be below zero, resetting to 0 is handled elsewhere.
+                    // However, we do the same check here to ensure 0 and below values are non-prioritized sources.
+                    if (priority <= 0)
+                    {
+                        repositories.Add(packageRepository);
+                    }
+                    else if (priorityRepositories.ContainsKey(priority))
+                    {
+                        priorityRepositories[priority].Add(packageRepository);
                     }
                     else
                     {
-                        repositories.Add(new DataServicePackageRepository(new RedirectedHttpClient(uri, bypassProxy) { UserAgent = "Chocolatey Core" }, packageDownloader) { Logger = nugetLogger });
+                        priorityRepositories.Add(priority, new List<IPackageRepository> { packageRepository });
                     }
                 }
                 catch (Exception)
                 {
-                    repositories.Add(new ChocolateyLocalPackageRepository(source) { Logger = nugetLogger });
+                    var packageRepository = new ChocolateyLocalPackageRepository(source) { Logger = nugetLogger };
+
+                    // The priority should never be below zero, resetting to 0 is handled elsewhere.
+                    // However, we do the same check here to ensure 0 and below values are non-prioritized sources.
+                    if (priority <= 0)

```

Comment:
I know it is being reset to zero in some location, need to figure out where so we possibly can output a warning as it could be considered a configuration issue.